### PR TITLE
New feature added to add12_6_4.py file.

### PR DESCRIPTION
**Pull request by pengfeili1**
Could better handle different situations. Adjust the code to easier to read.

### CODE REVIEW

**Comment by pengfeili1** [comment]
Hi Jason,

I have modified the code with renaming some variables, making them easier to read. Modifications have also been made to rise an error when there are two atom types have same VDW parameters but different polarizability parameters.

All the best,
Pengfei

**Comment by hainm** [comment]
It is likely you need to add a test for raising.

File: parmed/tools/add1264.py [hunk 1]
```diff
@@ -78,89 +78,88 @@ def params1264(parm, mask, c4file, watermodel, polfile, tunfactor):
...
-    # Determine the C4 term between the atom type of center metal ion 
-    # and every atom type in the prmtop file 
-
-    # 1. Get the list of AMBER_ATOM_TYPE and ATOM_TYPE_INDEX for all 
+        mettypind = parm.parm_data['ATOM_TYPE_INDEX'][i]
+        metchg = parm.parm_data['CHARGE'][i]
+        if mettypind in mettypdict: continue
+        mettypdict[mettypind] = (parm.atoms[i].atomic_number, int(metchg))
+        print "The selected metal ion is", \
+                pt.Element[parm.atoms[i].atomic_number]
```
Comment (swails) [inline parmed/tools/add1264.py]: This needs to be changed to:

```Python
        print("The selected metal ion is",
              pt.Element[parm.atoms[i].atomic_number])
```

We support both Python 2 and Python 3 in the same code base, and this won't work with Python 3.  You can put `from __future__ import print_function` at the top to get the right behavior.

File: parmed/tools/add1264.py [hunk 2]
```diff
@@ -78,89 +78,88 @@ def params1264(parm, mask, c4file, watermodel, polfile, tunfactor):
...
+    mettypinds = mettypdict.keys()
+    mettypinds.sort()
```
Comment (swails) [inline parmed/tools/add1264.py]: This doesn't work for Python 3, either.  Unlike Python 2, Python 3 does not return a list with `dict.keys()` -- it returns a more efficient `dict_keys` iterator that consumes less memory.

Change these two lines into a single line:

```Python
    mettypinds = sorted(mettypdict.keys())
```

**Comment by swails** [comment]
Ack!  Sorry -- I added tests for `add12_6_4`, but I didn't give them the right name so they're not run by default.  I've fixed that in #411, but what that will show is that this PR fails those tests (at least for Python 3).

I'll pull this PR into that PR and fix up the issues.  As @hainm said, if you have a test case for the failing test, then by all means go ahead and add it, but if not that's fine, too.

Thanks!

**Comment by swails** [comment]
OK.  @pengfeili1 -- your commit is included in the #411 PR, so I'm going to go ahead and close this.  Let me know if it doesn't work for you for whatever reason.

I'll also go ahead and update the version in the Amber git repo with these changes.

**Pull request closed without merging**